src/IO/LZMADeflatingWriteBuffer.h

```diff
@@ -14,22 +14,32 @@ namespace DB
 class LZMADeflatingWriteBuffer : public WriteBufferWithOwnMemoryDecorator
 {
 public:
+    template<typename WriteBufferT>
     LZMADeflatingWriteBuffer(
-        std::unique_ptr<WriteBuffer> out_,
+        WriteBufferT && out_,
```

Comment:
Is it necessary?

Reply:
`LZMADeflatingWriteBuffer` is inherited from `WriteBufferWithOwnMemoryDecorator` which is a specialization of `WriteBufferDecorator` and should now accommodate both owning and non-owning construction